### PR TITLE
Add TLD for lakeFS Simplified Authorization

### PR DESCRIPTION
**Pull request by arielshaqed**

### CODE REVIEW

File: design/open/acl/rbac-to-acl.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+Every group on the list of groups on the Groups page uses the format:
+
+| Group ID | Permission          | Created at | Repositories |
+|----------|---------------------|------------|--------------|
+| <name>   | Permission dropdown | datetime   | <number>     |
+
+![Groups page has a dropdown to edit permission for each
+group](./groups-with-perms.png)
+
+This gives information about the group and allows editing its permission.
```
Comment (ortz) [inline design/open/acl/rbac-to-acl.md:73]: One thing that I think is worth taking into account, we should block if someone is trying to remove the last admin group available (e.g deleting the group or changing its permissions, and essentially blocking themselves outside of lakefs management)

File: design/open/acl/rbac-to-acl.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+The upgrade script will _ensure_ that the 4 default global groups exist, and
+_modify_ existing groups to fit into the new ACLs scheme.
+
+* When creating the 4 default global groups: if another group exists and has
+  the desired name, upgrading will rename it by appending ".orig".  So after
+  upgrading the 4 default global groups exist, with these known names.
+* For any group, upgrading configured policies follows these rules, possibly
+  **increasing** access:
+
+  1. Any "Deny" rules are stripped, and a warning printed.
```
Comment (ortz) [inline design/open/acl/rbac-to-acl.md:118]: please make sure the warning message contains a link to our slack workspace or support email if they need any assistance.

File: design/open/acl/rbac-to-acl.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+<tr><td>
+```json
+{
+  "statement": [
+    {
+      "action": ["fs:List*", "fs:Read*"],
+      "effect": "allow",
+      "resource": "*"
+    }
+  ],
```
Comment (ortz) [inline design/open/acl/rbac-to-acl.md:158]: I wonder, is the UI going to be aware of this configuration? or you placed it here for the example and comparison between what the server is aware of and taking care of the translation?

**Review by ortz — APPROVED** [review]
Looks good, thank you!
some comments

File: design/open/acl/rbac-to-acl.md (hunk 1 above)
Comment (arielshaqed) [reply design/open/acl/rbac-to-acl.md:73]: Absolutely!  I suggest making the 4 "global" groups immutable and undeletable.

Admins will still be able to lock themselves out by removing all users from Admins.  The way out will be to run `lakefs superuser`.

WDYT?

File: design/open/acl/rbac-to-acl.md (hunk 3 above)
Comment (arielshaqed) [reply design/open/acl/rbac-to-acl.md:160]: The UI will use new APIs.  Those APIs will read and write (only) the UI part of the object.  OpenAPI will mirror the [ACL model in this PR](https://github.com/treeverse/lakeFS/pull/5297/files#diff-fa8533144f17ae3823634d8731349205bbe38a54930cf123241bca0c814ff412R153-R156).

**Review by arielshaqed — COMMENTED** [review]
Thanks!

Asking for a re-review to discuss how we prevent admins locking themselves out.

File: design/open/acl/rbac-to-acl.md (hunk 1 above)
Comment (ortz) [reply design/open/acl/rbac-to-acl.md:73]: sounds good, so the 4 global groups are associated with all repos and can't be modified, and if you want to scope users to specific repos you'll create new group?

**Review by ortz — APPROVED** [review]
🙏

File: design/open/acl/rbac-to-acl.md (hunk 1 above)
Comment (arielshaqed) [reply design/open/acl/rbac-to-acl.md:73]: I think that will be easiest.  This is very light enforcement; being able to modify these groups will be too confusing.

I still want to allow creating new groups with Admin ACL.  While all groups with this ACL have identical powers ("do anything"), someone might want to label their "admin" users in two groups for management reasons or something.

**Comment by arielshaqed** [comment]
@nopcoder I think you volunteered :-), but please remove yourself if you didn't...

**Comment by nopcoder** [comment]
yes, I'm still on it

File: design/open/acl/rbac-to-acl.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+}
+```
+</td></tr>
+</table>
+
+The UI receives and edits only the ui sub-object.  lakeFS translates it back
+to a statement when the UI posts it back.  In effect PBAC statements are the
+materialized form of the ui sub-object.
+
+## Milestones
```
Comment (nopcoder) [inline design/open/acl/rbac-to-acl.md:194]: How the transition will be - I mean we commit everything after all is tested outside main?
Do we plan to deploy some functionality and disable some as we go between milestones?

**Review by nopcoder — APPROVED** [review]
LGTM - couple of questions to understand better.

File: design/open/acl/rbac-to-acl.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+| **Super**  | Allows all operations except auth.         | SuperUsers (with changes) |
+| **Admin**  | Allows all operations.                     | Admins                    |
+
+In existing RBAC terms, the only permissions for non-admins are under `fs:*`
+and those to manage own credentials(`AuthManageOwnCredentials`).  The latter
+is given to all users.
+
+Some service jobs such as GC must be run using a user with Admin permission.
+
+##### Scopes
```
[the remark this answers is not in the document]
Comment (arielshaqed) [reply design/open/acl/rbac-to-acl.md:42]: Yes!

File: design/open/acl/rbac-to-acl.md [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+
+##### Permissions
+
+A _permission_ allows a group of users to do something.  Users cannot create
+permissions, lakeFS provides only these permissions:
+
+| Permission | Allows                                     | Existing Group            |
+|------------|--------------------------------------------|---------------------------|
+| **Read**   | Read operations, creating access keys.     | Viewers                   |
+| **Write**  | Allows all data read and write operations. | Developers                |
```
[the remark this answers is not in the document]
Comment (arielshaqed) [reply design/open/acl/rbac-to-acl.md:32]: No, sorry(?).  Same permissions  you'd get from FSReadWrite today.

File: design/open/acl/rbac-to-acl.md (hunk 4 above)
Comment (arielshaqed) [reply design/open/acl/rbac-to-acl.md:194]: No, the whole thing drops as a single release.  That release will delete P/RBAC and replace it with ACLs.

**Review by arielshaqed — COMMENTED** [review]
THANKS!

Pulling...

**Pull request merged**